pkg/cmd/target.go

```diff
@@ -413,21 +414,29 @@ func targetSeed(targetReader TargetReader, targetWriter TargetWriter, name strin
 }
 
 // resolveNameShoot resolves name to shoot
-func resolveNameShoot(target TargetInterface, name string) (matches []gardencorev1alpha1.Shoot) {
+func resolveNameShoot(target TargetInterface, name string) []gardencorev1alpha1.Shoot {
 	gardenClientset, err := target.GardenerClient()
 	checkError(err)
 
+	isRegexName := true
+	listOptions := metav1.ListOptions{}
+	if !strings.Contains(name, "*") {
+		isRegexName = false
+		fieldSelector := fields.OneTermEqualSelector("metadata.name", name)
+		listOptions.FieldSelector = fieldSelector.String()
+	}
+
 	var shootList *gardencorev1alpha1.ShootList
 	if len(target.Stack()) > 1 && target.Stack()[1].Kind == TargetKindProject {
 		projectName := target.Stack()[1].Name
 		project, err := gardenClientset.CoreV1alpha1().Projects().Get(projectName, metav1.GetOptions{})
 		checkError(err)
 
 		projectNamespace := project.Spec.Namespace
-		shootList, err = gardenClientset.CoreV1alpha1().Shoots(*projectNamespace).List(metav1.ListOptions{})
+		shootList, err = gardenClientset.CoreV1alpha1().Shoots(*projectNamespace).List(listOptions)
 		checkError(err)
 	} else if len(target.Stack()) > 1 && target.Stack()[1].Kind == TargetKindSeed {
-		shootList, err = gardenClientset.CoreV1alpha1().Shoots("").List(metav1.ListOptions{})
+		shootList, err = gardenClientset.CoreV1alpha1().Shoots(metav1.NamespaceAll).List(listOptions)
```

Comment:
in this case field selector `spec.cloud.seed=<seed-name>` can be used, for details see https://github.com/gardener/gardener/pull/677

Reply:
```
Error from server (BadRequest): Unable to find "core.gardener.cloud/v1alpha1, Resource=shoots" that match label selector "", field selector "spec.seedName=aws": "spec.seedName" is not a known field selector: only "metadata.name", "metadata.namespace"
```

Looks like this feature is gone with the new apiVersion. Will check how to fix it in g/g.

Reply:
Btw can we proceed without the fieldSelector by seedName as it will take time until g/g is released and etc.?

Reply:
yes, sure.
I was not aware that this feature was not migrated to the new API